test/unit/Cardano/Wallet/Primitive/ModelSpec.hs

```diff
@@ -127,11 +133,19 @@ prop_applyBlockBasic s =
             (wallet :| _) = foldl (flip applyBlock) checkpoints blockchain
             utxo = totalUTxO wallet
             utxo' = evalState (foldM (flip updateUTxO) mempty blockchain) s
+            txMeta = getTxMetas wallet
         in
             (ShowFmt utxo === ShowFmt utxo') .&&.
             (availableBalance wallet === balance utxo') .&&.
             (totalBalance wallet === balance utxo') .&&.
-            (ShowFmt (getTxMetas wallet) === ShowFmt mempty)
+
+            -- TxMeta properties ... fairly trivial properties
+            (cond0 ==> (
+              (Set.map direction txMeta === Set.singleton Incoming) .&&.
+              (Set.map status txMeta === Set.singleton InLedger) .&&.
+              (Set.findMin (Set.map txMetaSlotId txMeta) >=? SlotId 14 0) .&&.
+              (Set.findMax (Set.map txMetaSlotId txMeta) <=? SlotId 14 19)
+            ))
```

Comment:
Hmm.. I'd rather have that in another property. It's good to keep tests focused on testing one and one thing. Here, it's really about comparing the UTxO.

Also, I don't get the first and two last assertions; `Set.map direction txMeta === Set.singleton Incoming` _happens_ to be true on our testing data but isn't a property of tx metas in general :confused: